test/path.jl

```diff
@@ -9,6 +9,23 @@
         @test isabspath(S(homedir()))
         @test !isabspath(S("foo"))
     end
+    if Sys.iswindows()
+        @testset "issue #38491" begin
+            pwd_drive = uppercase(splitdrive(pwd())[1])
+            drive = if (pwd_drive != "X:") "X:" else "Y:" end
```

Comment:
```suggestion
            drive = (pwd_drive == "X:") ? "Y:" : "X:"
```
optional style nit: this is a somewhat more canonical form